src/main/java/de/tum/in/www1/artemis/domain/participation/Participation.java

```diff
@@ -116,6 +116,17 @@ public abstract class Participation extends DomainObject implements Participatio
     @Cache(usage = CacheConcurrencyStrategy.NONSTRICT_READ_WRITE)
     private Set<Submission> submissions = new HashSet<>();
 
+    /**
+     * Graded course exercises, practice mode course exercises and real exam exercises always have only one parcitipation per exercise
+     * In case of a test exam, there are multiple participations possible for one exercise
+     * This field is necessary to preserve the constraint of one partipation per exercise, while allowing multiple particpiations per exercise for text exams
+     * The value is 0 for graded course exercises and exercises in the real exams
+     * The value is 1 for practice mode course exercises
+     * The value is 0-255 for test exam exercises. For each subsequent participation the number is increased by one
+     */
+    @Column(name = "number_of_attempts")
+    private int numberOfAttempts = 0;
```

Comment:
NumberOfAttempts sounds to me like the maximum number of attempts, but the JavaDoc reads more like this is an indicator on which attempt this is. So just `attempt` would be a more fitting name imo.